rfcs/convergence/authoring-stories.md

```diff
@@ -267,13 +270,40 @@ TBA
 
 TBA
 
-### 7. how structure stories in the storybook nav tree
+### 7. dissecting big story files into smaller ones
 
-TBA
+Components with complex API require many long stories, both for documentation and for e2e testing. This creates long files with hundres of lines, which are hard to maintain. How can these files be dissected into smaller ones?
+
+**Proposal:**
 
-### 8. dissecting big story files into smaller ones
+1. Every component can have at most one `.stories.tsx` file. This file has a default export with Storybook configuration for the docs page.
+2. If putting all stories into one file would make it too long, individual stories might be put into `.story.tsx` files as a named export, and then re-exported from `.stories.tsx` file like this: `export * from ‘./IndividualStoryFile.story’;`
+3. If individual story files are employed, `.stories.tsx` file must not contain any stories.
 
-TBA
+### 8. location and naming convention
+
+1. Story files should be in the same folder as the component which they are targeting.
+2. Name of every .stories.tsx / .story.tsx file should start with name of a component which it is targeting, to improve colocation in alphabetical file ordering. If appropriate, story name should be adjusted via configuration to a well readable name, suitable for documentation, like this:
+
+```tsx
+export const ButtonPrimary = (props: ButtonProps) => <Button {...props}>Text</Button>;
+ButtonPrimary.args = {
+  primary: true,
+};
+ButtonPrimary.storyName = 'Better story name';
+```
+
+### 9. default story
+
+Every documented component must have a story called `Default`. This story must support manipulation with [Controls](https://storybook.js.org/docs/react/essentials/controls). Ordering of stories will be set up, so this would be the first story in Storybook nav tree.
+
+### 10. design of stories
+
+Public stories should follow Fluent Design Language to give developers better feel for patterns they should utilise. For example, when a button is necessary do demonstrate usage of a component, Fluent UI Button should be used instead of a pure HTML button.
+
+### 11. story code must be useful
+
+Public stories should contain only code, which is useful for users to see after clicking on “Show code” in documentation. Extra markup (e.g., container with CSS styles) can be added via [Decorators](https://storybook.js.org/docs/react/writing-stories/decorators).
```

Comment:
Should we explicitly duplicate code to fulfil this requirement ? otherwise users would see `XYZHelper` in a bunch of stories that wouldn't mean anything

Reply:
I am not sure if I understand your concern well. You shouldn’t duplicate code.

If the decorator is useful for all stories on the documentation page, you should specify it in the default export:
```jsx
export default {
  title: 'Button',
  component: Button,
  decorators: [
    (Story) => (
      <div style={{ margin: '3em' }}>
        <Story />
      </div>
    ),
  ],
};
```

If the decorator is useful for multiple stories which differ only in props, you should use `.bind()`:
```jsx
export const ButtonStory = …
ButtonStory.decorators = [(Story) => <div style={{ margin: '3em' }}><Story/></div>];

const Primary = ButtonStory.bind({});
Primary.args = {
  primary: true,
  label: 'Button',
}

const Secondary = ButtonStory.bind({});
Secondary.args = {
  ...Primary.args,
  primary: false,
}
```

If you can’t use `bind()` you should create a function which would be used in both stories. The documentation users will not see the usage of the function.

Do you see any issue with this?

Reply:
I can’t think of any useful guidance for that scenario. It’s at the discretion of the developer to consider trade offs between readability of code, appealing appearance and proper interactivity of the story.

If you were thinking of configuring the code manually with `Story.parameters.docs.source.code`, I don’t think it’s a good idea because it would break live code editing.

Can you think of any guidance that would be useful?

Reply:
> If you were thinking of configuring the code manually w
nope that's not what I thought 

following code
```
function doFoo(){...}
function doBar(){...}

export const MyStory = (args) => {
   return <div>
      <Button>click</Button>
      doFoo().map(render....)
     {dooBar() ? 'hello' : <div>nope</div> }
  </div>
}
```

will show something like this in that `Show code` panel

```
<div>
      <Button>click</Button>
      doFoo().map(render....)
     {dooBar() ? 'hello' : <div>nope</div> }
  </div>
```

which will confuse the user as he has no idea what those functions do

Reply:
After discussion with @Hotell, we agreed that it’s better not to use `.bind()`, as it creates unnecessary cognitive overload. Exporting decorator to a function is easier to read, so this is the preferred approach.

```jsx
const buttonDecorator = (Story) => <div style={{ margin: '3em' }}><Story/></div>;

const Primary = <Button primary="true">Primary</Button>
Primary.decorators = [buttonDecorator];

const Secondary = <Button>Secondary</Button>
Secondary.decorators = [buttonDecorator];